compiler/ballerina-lang/src/main/java/io/ballerina/projects/internal/ProjectFiles.java

```diff
@@ -147,13 +149,18 @@ private static DocumentData loadTestDocument(Path documentFilePath) {
         return DocumentData.from(ProjectConstants.TEST_DIR_NAME + "/" + documentName, content);
     }
 
-    public static PackageManifest createPackageManifest(Path ballerinaTomlFilePath) {
-        return BallerinaTomlProcessor.parseAsPackageManifest(ballerinaTomlFilePath);
+    public static PackageManifest createPackageManifest(BallerinaToml ballerinaToml) {
+        return ballerinaToml.packageManifest();
     }
 
     public static BuildOptions createBuildOptions(Path projectPath, BuildOptions theirOptions) {
         Path ballerinaTomlFilePath = projectPath.resolve(ProjectConstants.BALLERINA_TOML);
-        BuildOptions defaultBuildOptions = BallerinaTomlProcessor.parse(ballerinaTomlFilePath).getBuildOptions();
+        BallerinaToml ballerinaToml = BallerinaToml.from(ballerinaTomlFilePath);
+        if (ballerinaToml.diagnostics().hasErrors()) {
+            throw new BallerinaTomlException(ballerinaToml.getErrorMessage());
```

Comment:
Should we throw the exception here? IMO, we should add it to TomlDiagnostics and continue the execution.